File: go/exercise05/simple_math.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	fmt.Print("What is the first number? ")
	scanner.Scan()
	firstNumber, _ := strconv.ParseFloat(scanner.Text(), 64)

	fmt.Print("What is the second number? ")
	scanner.Scan()
	secondNumber, _ := strconv.ParseFloat(scanner.Text(), 64)

	sum := add(firstNumber, secondNumber)
	difference := subtract(firstNumber, secondNumber)
	product := multiply(firstNumber, secondNumber)
	quotient := divide(firstNumber, secondNumber)

	fmt.Printf("%v + %v = %v\n", firstNumber, secondNumber, sum)
	fmt.Printf("%v - %v = %v\n", firstNumber, secondNumber, difference)
	fmt.Printf("%v * %v = %v\n", firstNumber, secondNumber, product)
	fmt.Printf("%v / %v = %v\n", firstNumber, secondNumber, quotient)
}

func add(a, b float64) float64 {
	return a + b
}

func subtract(a, b float64) float64 {
	return a - b
}

func multiply(a, b float64) float64 {
	return a * b
}

func divide(a, b float64) float64 {
	if b == 0 {
		return 0
	}
	return a / b
}

File: python/exercise05/simple_math.py
def add(a: float, b: float) -> float:
    return a + b


def subtract(a: float, b: float) -> float:
    return a - b


def multiply(a: float, b: float) -> float:
    return a * b


def divide(a: float, b: float) -> float:
    if b == 0:
        return 0

    return a / b


first_number: int = int(input("What is the first number? "))
second_number: int = int(input("What is the second number? "))

sum = add(first_number, second_number)
difference = subtract(first_number, second_number)
product = multiply(first_number, second_number)
quotient = divide(first_number, second_number)

print("%v + %v = %v\n", first_number, second_number, sum)
print("%v - %v = %v\n", first_number, second_number, difference)
print("%v * %v = %v\n", first_number, second_number, product)
print("%v / %v = %v\n", first_number, second_number, quotient)
